test/SILOptimizer/copyforward_ossa.sil

```diff
@@ -0,0 +1,845 @@
+// RUN: %target-sil-opt  -enable-sil-verify-all %s -copy-forwarding -enable-copyforwarding -enable-destroyhoisting | %FileCheck %s
```

Comment:
Here's what I did which converting the test to OSSA 
- Update tests to obey OSSA rules
- Removed tests with address phi args: Ex: dont_assert_nrvo, dont_assert_on_basic_block_arg, dont_assert_backward_init have copy_addr with address phi operands
- testExtractedStoredValueUser1 and testExtractedStoredValueUser2 differ only w.r.t strong_retain and retain_value retain only one of them
- Update tests to remove debug_value_addr on uninitialized memory
- Added rdars for cleanup of debug_value_addr after forwardPropgate and backwardPropagate
- Some tests were forced to be written in OSSA in such a way that there was no way to test the original bug. I have retained these tests even though they are not testing any past bug anymore: Ex: foo
- Update test so that bug can still be tested: Ex: testLoadDestroy

Reply:
That all sounds very reasonable to me.